subcommands/keys/ca_revoke_device_ca.go

```diff
@@ -0,0 +1,224 @@
+package keys
+
+import (
+	x509Lib "crypto/x509"
+	"encoding/asn1"
+	"encoding/hex"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"math/big"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/foundriesio/fioctl/client"
+	"github.com/foundriesio/fioctl/subcommands"
+	"github.com/foundriesio/fioctl/x509"
+)
+
+const (
+	crlCmdAnnotation      = "crl-revoke"
+	crlCmdRevokeDeviceCa  = "revoke-device-ca"
+	crlCmdDisableDeviceCa = "disable-device-ca"
+)
+
+func init() {
+	revokeCmd := &cobra.Command{
+		Use:   "revoke-device-ca <PKI Directory>",
+		Short: "Revoke device CA from the list of CAs allowed to issue device client certificates",
+		Run:   doRevokeDeviceCa,
+		Args:  cobra.ExactArgs(1),
+		Long: `Revoke device CA from the list of CAs allowed to issue device client certificates.
+
+When the online or local device CA is revoked:
+- It is no longer possible to register new devices with client certificates issued by this device CA.
+- Existing devices with client certificates issued by this device CA can no longer connect to your FoundriesFactory.
+
+You may later re-add a revoked device CA using "keys ca update", if you still keep its certificate stored somewhere.
+Once you do it, devices with client certificates issued by this device CA may connect to your FoundriesFactory again.`,
+		Annotations: map[string]string{crlCmdAnnotation: crlCmdRevokeDeviceCa},
+	}
+	caCmd.AddCommand(revokeCmd)
+	addRevokeCmdFlags(revokeCmd, "revoke")
+
+	disableCmd := &cobra.Command{
+		Use:   "disable-device-ca <PKI Directory>",
+		Short: "Disable device CA devices with client certificates issued by it can no longer be registered",
+		Run:   doRevokeDeviceCa,
+		Args:  cobra.ExactArgs(1),
+		Long: `Disable device CA devices with client certificates issued by it can no longer be registered.
+
+When the online or local device CA is disabled:
+- It is no longer possible to register new devices with client certificates issued by this device CA.
+- Existing devices with client certificates issued by this device CA can continue to connect and use FoundriesFactory.
+
+Usually, when the device CA is compromised, a user would:
+1. Immediately disable a given device CA using "fioctl keys ca disable-device-ca <PKI Directory> --serial <CA Serial>".
+2. Inspect their devices with client certificates issued by that device CA, and remove any compromised devices.
```

Comment:
Can we add here the command to remove the device as well?
I loved the steps that make the feature much more clear.

Reply:
Then I'd have to also add a command to list devices and a command to inspect individual device.
Though I think it makes sense to add smth like `(see "fioctl devices")` at the end.

Reply:
`fioctl devices delete`

Reply:
The use needs to have a reference how to "inspect" and how to "delete"; `fioctl devices` covers both necessities.

Reply:
https://github.com/foundriesio/fioctl/pull/324/commits/8eb792f387f318827e7bb98d90b1f0a31355addc